### PR TITLE
Add testing main concept.

### PR DESCRIPTION
**Pull request by xavierlepretre**
This PR introduces testing in Cosmos as a general concept.

### Change scope

The changes in this Pull Request include (please tick all that apply):

* [ ] Small language/grammar fixes
* [ ] Small content fixes 
* [x] Addition of new content
* [ ] Sample code/command updates
* [ ] Platform fixes
* [ ] Other

### CODE REVIEW

File: academy/2-cosmos-concepts/11-testing.md
Comment (julienrbrt) [inline academy/2-cosmos-concepts/11-testing.md:60]:
```suggestion
This is why from version 0.47 of the Cosmos SDK, each module exposes functions to provision a minimally viable module. This way, your module only knows to call the right functions in its dependent module but no further down the line. For instance, when (integration-)testing the bank module, no slashing module is provisioned as it is a side concern instead of a dependency. To facilitate these integration tests, the Cosmos team has also developed an in-house dependency injection. Note that integration tests do not create an application, but instead a chain of dependencies.
```
The bank module is a wrong example, as we require staking. But this is indeed true.
Integration tests do create an application, the application just have the least module enabled.

**Review by julienrbrt — APPROVED** [review]
lgtm!

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [reply academy/2-cosmos-concepts/11-testing.md:60]: Yes, it is (1).
Yes, it continues until all dependencies are satisfied.
This may mean that in an app made with "spaghetti code", you pull everything and there are no differences between an MVA and a full app...

Comment (AndrewB9lab) [reply academy/2-cosmos-concepts/11-testing.md:60]: Thanks!

**Comment by coldice** [comment]
ready for merge?

**Comment by xavierlepretre** [comment]
> ready for merge?

Let's wait for @AndrewB9lab To formally approve it 😏

**Review by AndrewB9lab — APPROVED** [review]
Language review complete.

**Comment by xavierlepretre** [comment]
> Language review complete.

@CitMC Your turn?

File: academy/2-cosmos-concepts/11-testing.md
Comment (CitMC) [inline academy/2-cosmos-concepts/11-testing.md:106]: @xavierlepretre please add a synopsis. After, this PR can be merged.

**Review by CitMC — COMMENTED** [review]
After a synopsis box is added, this section can be merged.

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:60]:
```suggestion
A minimum-viable application contains your module and all its dependencies, as well as their dependencies, but nothing more.
```

Comment (xavierlepretre) [reply academy/2-cosmos-concepts/11-testing.md:60]: Also you reverted my succinct way, after Andrew had disambiguated the meaning.

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:66]: > ...on each module exposes functions...
Does not sound right to me.

```suggestion
This is why, to minimize correlations, from version 0.47 of the Cosmos SDK on each module exposes functions to provide a minimum viable module. This way, your module only knows how to instantiate itself given fully-fledged dependencies, the _inputs_. An added benefit is that your module exposes explicitly the inputs it needs to instantiate.
```

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:68]:
```suggestion
Your integration tests start by creating an app that instantiates the list of explicitly defined inputs required. For instance, when creating a minimum-viable app to integration-test the bank module, no slashing module is provisioned, as slashing is a side concern instead of a dependency. Of course, each module requires a different minimum-viable app. To facilitate the creation of such an app and therefore, of integration tests, the Cosmos SDK team has also developed an in-house dependency injection.
```

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:74]:
```suggestion
In Cosmos SDK version 0.46 and earlier, what are called "integration tests" are really full tests, where a full application is being instantiated. The reason behind this is that in these versions the coding effort to create a minimum-viable app was not commensurate with the benefit it provided compared to a full app.
```

File: academy/2-cosmos-concepts/11-testing.md
Comment (AndrewB9lab) [reply academy/2-cosmos-concepts/11-testing.md:66]: Could change to "...from version 0.47 of the Cosmos SDK onwrd each module exposes..."

OR, remove "on" completely.

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:82]:
```suggestion
Simulation tests are similar in scope to integration tests, where they reuse your module's minimum-viable application. This scope also only starts to make sense from Cosmos SDK 0.47, where the application dependencies are disentangled.
```

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [inline academy/2-cosmos-concepts/11-testing.md:86]:
```suggestion
End-to-end (E2E) tests are at the top of the testing pyramid. Unlike integration and simulation tests, they work by testing the _full_ application, not a minimum-viable one.
```

File: academy/2-cosmos-concepts/11-testing.md
Comment (xavierlepretre) [reply academy/2-cosmos-concepts/11-testing.md:66]: aah on was for onward

Comment (CitMC) [reply academy/2-cosmos-concepts/11-testing.md]: Changed. Thank you Xavier and Andrew! :)

**Review by coldice — APPROVED** [review]

**Pull request merged**
